saleor/core/management/commands/fix_undiscounted_total_prices.py

```diff
@@ -0,0 +1,157 @@
+        Exists(
+            OrderLine.objects.filter(
+                order_id=OuterRef("id"), voucher_code__isnull=False
+            )
+        ),
+        total_gross_amount=F("undiscounted_total_gross_amount"),
+    ).order_by("id")
+
+    updated_orders_pks = []
+    for batch_pks in queryset_in_batches(orders_to_update):
+        orders = Order.objects.filter(pk__in=batch_pks)
+        lines = OrderLine.objects.filter(order_id__in=orders.values("id")).values(
+            "order_id",
+            "undiscounted_total_price_gross_amount",
+            "total_price_gross_amount",
+            "undiscounted_total_price_net_amount",
+            "total_price_net_amount",
+        )
+        lines_discount_data = defaultdict(lambda: (0, 0))
+        for data in lines:
+            discount_amount_gross = (
+                data["undiscounted_total_price_gross_amount"]
+                - data["total_price_gross_amount"]
+            )
+            discount_amount_net = (
+                data["undiscounted_total_price_net_amount"]
+                - data["total_price_net_amount"]
+            )
+            current_discount_gross, current_discount_net = lines_discount_data[
+                data["order_id"]
+            ]
+            lines_discount_data[data["order_id"]] = (
+                current_discount_gross + discount_amount_gross,
+                current_discount_net + discount_amount_net,
+            )
+
+        for order in orders:
+            discount_amount_gross, discount_amount_net = lines_discount_data.get(
+                order.id
+            )
+            if discount_amount_gross > 0 or discount_amount_net > 0:
+                order.undiscounted_total_gross_amount += discount_amount_gross
+                order.undiscounted_total_net_amount += discount_amount_net
+
+                updated_orders_pks.append(order.id)
+
+        Order.objects.bulk_update(
+            orders,
+            [
+                "undiscounted_total_gross_amount",
+                "undiscounted_total_net_amount",
+            ],
+        )
+    return set(updated_orders_pks)
+
+
+def send_order_updated_events(order_pks):
+    list_pks = list(order_pks)
+
+    for index in range(0, len(list_pks), SEND_ORDER_UPDATED_BATCH_SIZE):
+        send_order_updated.delay(
+            list_pks[index : index + SEND_ORDER_UPDATED_BATCH_SIZE]
+        )
+
+
+@app.task
+def fix_undiscounted_prices():
+    # 0139_fix_undiscounted_total_on_lines
+    with connection.cursor() as cursor:
+        cursor.execute(RAW_0139_UPDATE_SQL)
+        records = cursor.fetchall()
+    updated_orders_pks_from_lines = set(
+        [record[0] for record in records] if records else []
+    )
+
+    # 0140_fix_order_undiscounted_total
+    updated_orders_pks_from_orders = update_order_undiscounted_price()
+
+    # send webhooks for unique order ids
+    if all_order_pks := updated_orders_pks_from_lines | updated_orders_pks_from_orders:
```

Comment:
I don't think variables `updated_orders_pks_from_lines` and `updated_orders_pks_from_orders` needed separately since at the end you need just combined. Could you create one variable for this?

Reply:
I am, right on this line. Both ID lists come from different queries so we combine them together to shoot hooks only for unique ones. We delete partial sets afterward to save memory.